docs/proposals/hpa-custom-metrics.md

```diff
@@ -0,0 +1,162 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<!-- TAG RELEASE_LINK, added by the munger automatically -->
+<strong>
+The latest release of this document can be found
+[here](http://releases.k8s.io/release-1.3/docs/proposals/custom-metrics.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+Specifying Metrics in the HPA
+=============================
+
+Introduction
+------------
+
+Currently, custom metrics specified via a list of metric names
+and target values.  Each metric name is prefixed with "custom/",
+and then is requested at the pod level for each pod of the target
+scalable.
+
+Several use cases are not supported by this design:
+
+### Refering to Non-Custom Metrics ###
+
+Several users have asked for the ability to scale on existing built-in metrics
+(e.g. memory usage).  While the current design does allow the user to specify
```

Comment:
A valid use case for memory-based autoscaling are Data Grids (where you spread your data across the cluster). Typically when you add a new node, the cluster performs rebalancing procedure and the average memory usage will drop. 

Of course it requires proper configuration (e.g. setting `-Xms` in Java [[1](https://developerblog.redhat.com/2014/07/15/dude-wheres-my-paas-memory-tuning-javas-footprint-in-openshift-part-1/)])